node_modules/oae-authentication/lib/api.js

```diff
@@ -307,12 +308,19 @@ var _getOrCreateUser = function(ctx, loginId, displayName, opts, callback) {
                     opts.emailVerified = true;
                 }
 
-                createUser(ctx, loginId, displayName, opts, function(err, user) {
+                var shouldCheckEmail = (opts.email && !opts.authoritative);
+                OaeUtil.invokeIfNecessary(shouldCheckEmail, validateEmailBelongsToTenant, ctx, opts.email, function(err) {
```

Comment:
I think if a non-authoritative authentication method doesn't provide an email (e.g., Twitter), we should still restrict login if the email domain of the tenant is set. Otherwise, this would mean that the entire public can come in VIA twitter, which we're trying to avoid.